src/Microsoft.Framework.TestHost/Program.cs

```diff
@@ -34,11 +34,110 @@ public int Main(string[] args)
+                        if (message.MessageType == "TestDiscovery.Start")
+                        {
+                            var commandArgs = new List<string>()
+                            {
+                                "test",
+                                "--list",
+                                "--designtime"
+                            };
+
+                            var testServices = TestServices.CreateTestServices(_services, project, channel);
+                            await ProjectCommand.Execute(testServices, project, commandArgs.ToArray());
+
+                            Trace.TraceInformation("[ReportingChannel]: OnTransmit(DiscoverTests)");
+                            channel.Send(new Message()
+                            {
+                                MessageType = "TestDiscovery.Response",
+                            });
+                            return 0;
+                        }
+                        else if (message.MessageType == "TestExecution.Start")
+                        {
+                            var commandArgs = new List<string>()
+                            {
+                                "test",
+                                "--designtime"
+                            };
+
+                            var tests = message.Payload?.ToObject<RunTestsMessage>().Tests;
+                            if (tests != null)
+                            {
+                                foreach (var test in tests)
+                                {
+                                    commandArgs.Add("--test");
+                                    commandArgs.Add(test);
+                                }
+                            }
+
+                            var testServices = TestServices.CreateTestServices(_services, project, channel);
+                            await ProjectCommand.Execute(testServices, project, commandArgs.ToArray());
+
+                            Trace.TraceInformation("[ReportingChannel]: OnTransmit(ExecuteTests)");
+                            channel.Send(new Message()
+                            {
+                                MessageType = "TestExecution.Response",
+                            });
+                            return 0;
+                        }
+                        else
+                        {
+                            var error = string.Format("Unexpected message type: '{0}'.", message.MessageType);
+                            Trace.TraceError(error);
+                            channel.SendError(error);
+                            return -1;
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Trace.TraceError(ex.ToString());
+                        channel.SendError(ex);
+                        return -2;
+                    }
+                }
             });
```

Comment:
This code is copy-pasted from other placed in this class. Not refactoring because we need to delete the 'other places' after we implement this support in VS and give everyone a chance to upgrade.